core/src/test/java/org/apache/brooklyn/util/core/xstream/XmlUtilTest.java

```diff
@@ -20,15 +20,65 @@
 
 import static org.testng.Assert.assertEquals;
 
-import org.apache.brooklyn.util.core.xstream.XmlUtil;
+import org.apache.brooklyn.util.core.xstream.XmlUtil.Escaper;
 import org.testng.annotations.Test;
 
+import com.google.common.base.Optional;
+
 
 public class XmlUtilTest {
 
     @Test
     public void testXpath() throws Exception {
         String xml = "<a><b>myb</b></a>";
-        assertEquals(XmlUtil.xpath(xml, "/a/b[text()]"), "myb");
+        for (int i = 0; i < 2; i++) {
```

Comment:
One of the commits changes `XmlUtil` to use thread-local storage (rather than re-creating the `DocumentBuilder` on every invocation). By calling it twice, we confirm that thread-local storage isn't messing things up (e.g. if you removed the `documentBuilder.reset()` then it would probably break).